synapse/storage/databases/main/events_worker.py

```diff
@@ -231,10 +256,9 @@ def get_events(
 
         return {e.event_id: e for e in events}
 
-    @defer.inlineCallbacks
-    def get_events_as_list(
+    async def get_events_as_list(
         self,
-        event_ids: List[str],
+        event_ids: Iterable[str],
```

Comment:
We iterate over this multiple, so it can't be an iterable:

```suggestion
        event_ids: Collection[str],
```